Test_Cases.py

```diff
@@ -1,169 +1,216 @@
+
+        acceptance criteria
+        --------------------
+        -When an 8 character password is entered, the data error status
+         and the error text exhibit's this is too short of a password.
+
+        Note
+        --------------------
+        -Yahoo security will randomly ask to "prove you are not a robot".
+         Which, at times, can be responsible for the tests in Error Message
+         Passwords class to error. There is not an easy fix for this.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(5))
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "WEAK_PASSWORD")
+        self.assertEqual(yahoo_page.short_password_error_message().text,
+                         "- Your password is too easy to guess.")
+
+    def test_moderate_password_error_message(self):
+        """
+        Confirm that when the password 9 or 10 characters in length, the
+        error data status is "ALMOST_THERE". Confirm the resulting error
+        text is "- Almost there".
+
+        acceptance criteria
+        --------------------
+        -When a password that is between 9 and 10 characters is entered,
+         the data error status and the error text exhibit's this is a
+         moderate length for the password.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(9))
+        time.sleep(0.75)  # Please look at the branch Explicit_Wait_ErrorMessagePasswords if you want to see why I used this implicit wait. This keeps my code clean and readable. No E.C.s exist that can wait for this dynamic element correctly for this specific test. So I created my own E.C; which you can see being used in the given branch. This E.C. is also in Tools in this repo.
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "ALMOST_THERE")
+        self.assertEqual(yahoo_page.moderate_password_error_message().text,
+                         "- Almost there.")
+
+    def test_long_password_error_message(self):
+        """
+        Confirm that when the password is 11 characters, the error data
+        status is "STRONG_PASSWORD". Confirm the resulting error text is
+        "- you did it!".
+
+        acceptance criteria
+        --------------------
+        -When a password that is 11 characters is entered, the data
+         error and error text either does not exist or it exhibits this
+         is a password of sufficient length.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(11))
+        time.sleep(1)  # The Explicit_Wait_ErrorMessagePasswords branch has some really cool stuff in it. But if you look at it you will see why I decided to use this implicit wait instead
```

Comment:
Ditto in line 78.

I just want to add that even if this line of code is meant to be a comment that you're going to delete in the future, try to keep it shorter than 80 or 100 characters (depending on the company you end working for, they might even allow lines up to 110 chars long). This is to prevent these long lines showing in multiple lines in Github, making more difficult peer reviews and horizontal scrolling when reading your code in the local IDE 😉 

You could always have a multi-line comment in the code, such as the one you did in line 110-112.